db/migrate/20150405012126_remove_approval_status_from_reservation.rb

```diff
@@ -0,0 +1,35 @@
+class RemoveApprovalStatusFromReservation < ActiveRecord::Migration
+  def change
+    Reservation.all.each do |res|
+      if res.approval_status == 'approved'
+        res.flag(:request)
+      elsif res.approval_status == 'missed_and_emailed'
+        res.flag(:missed_email_sent)
+      end
+
+      if res.checked_in 
+        res.status = 'returned'
+        if res.checked_in > res.due_date
+          res.overdue = true
+        end
+      elsif res.checked_out
+        res.status = 'checked_out'
+        if res.due_date < Time.zone.today
+          res.overdue = true
+        end
+      elsif res.start_date < Time.zone.today
+        res.status = 'missed'
+      elsif res.approval_status == 'requested'
+        res.flag(:request)
+        res.status = 'requested'
+      elsif res.approval_status == 'denied'
+        res.flag(:request)
+        res.status = 'denied'
+      else
+        res.status = 'reserved'
+      end
```

Comment:
Don't forget to save the reservations. We should wrap the whole thing in a `Transaction` block so that we only modify the database if all of the records save correctly, otherwise we might to massage some data before we change *every* reservation in the table.